src/classiceditorui.js

```diff
@@ -89,6 +104,35 @@ export default class ClassicEditorUI {
 				}
 
 				return Promise.all( promises );
+			} )
+			.then( () => {
+				const toolbarFocusTracker = this.view.toolbar.focusTracker;
+
+				// Because toolbar items can get focus, the overall state of
+				// the toolbar must also be tracked.
+				this.focusTracker.add( this.view.toolbar.element );
+
+				// Listen on the keystrokes from the main UI.
+				this.keystrokes.listenTo( this.view.element );
+
+				// Listen on the keystrokes from the floating panels, toolbars and the such.
+				this.keystrokes.listenTo( this.view._bodyCollectionContainer );
+
+				// Focus the toolbar on the keystroke, if not already focused.
+				this.keystrokes.set( 'alt + f10', ( data, cancel ) => {
+					if ( this.focusTracker.isFocused && !toolbarFocusTracker.isFocused ) {
+						this.view.toolbar.focus();
+						cancel();
+					}
+				} );
+
+				// Blur the toolbar and bring the focus back to editable on the keystroke.
+				this.keystrokes.set( 'esc', ( data, cancel ) => {
```

Comment:
`ToolbarView` should never know about the environment it lives in except for what is passed in the constructor and what API it provides itself (observables). So to handle <kbd>Esc</kbd> and return the focus somewhere, it would cross boundaries it should have never known about. 

It is up to EditorUI to detect that keystroke, analyze the state of toolbar focus (it has own FTracker), consider the state of the editor and take the action if it makes sense.

> If you'll agree with me that editor UI doesn't need its own keystroke handler

If you read my answer above, you know that I disagree. Editing keystroke handler should not have any interest in navigation across UI element because it has nothing to do with the content. And vice versa, the EditorUI keystroke handler should never interfere with content creation processes. I think about these two as completely separate areas of the editor and I would rather not migle them.